design/metadata-handling.md

```diff
@@ -421,6 +430,14 @@ ways. The following types of objects are available and accept lists:
   `192.168.0.1` for a subnet `192.168.0.0/24`).
 - **fromHostInterfaces**: renders the MAC address of the BareMetalHost that
   matches the name given as value.
+- **fromLabels**: renders the content of a label on an object or an empty string
+  if the label is absent. It takes an `object` attribute to specify the type of
+  the object where to fetch the label, and a `label` attribute that contains the
+  label key.
+- **fromAnnotations**: renders the content of a annotation on an object or an
+  empty string if the annotation is absent. It takes an `object` attribute to
+  specify the type of the object where to fetch the annotation, and an
+  `annotation` attribute that contains the annotation key.
```

Comment:
What are the possible values for the object field? Can we define an enum?

Reply:
In the implementation of the current feature, it is an enum. I'll just document it here in the proposal : "machine, metal3machine, baremetalhost" (case insensitive)